### PR TITLE
11237 add ga events

### PR DESCRIPTION
**Pull request by cohnjesse**
## Description
[Original Ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/11237)

This PR is to add the GA event names that were provided to us by the Analytics team

## Testing done
existing unit tests still pass

## Acceptance criteria
- [x] Code/event names provided by the GA/Insights Team has been inserted into our feature for tracking

### CODE REVIEW

File: src/applications/disability-benefits/view-payments/actions/index.js [hunk 1]
```diff
@@ -19,8 +22,15 @@ export const getAllPayments = () => async dispatch => {
     // TODO: fire off analytics event when endpoint is wired up.
     //   const errCode = res.errors[0].code;
     //   isServerError(errCode) ? recordEvent({}) : recordEvent({})
+    recordEvent({
+      event: `disability-view-dependents-load-failed`,
```
Comment (micahchiang) [inline src/applications/disability-benefits/view-payments/actions/index.js:26]: This should be `view-payment-history-failed`

File: src/applications/disability-benefits/view-payments/actions/index.js [hunk 2]
```diff
@@ -19,8 +22,15 @@ export const getAllPayments = () => async dispatch => {
...
+      'error-key': `${response.errors[0].status}_description_of_error`,
```
Comment (micahchiang) [inline src/applications/disability-benefits/view-payments/actions/index.js:27]: I think this key is looking for an actual descriptor as opposed to the string `description_of_error`. Even though we haven't hooked redux up to the backend yet, we can probably still add those in and keep the messages general, similar to what's done in rated disabilities.

**Review by micahchiang — CHANGES_REQUESTED** [review]
Requesting change for the view-payment-history-failed event key. Also left a thought about adding in messages for the error key, even though we aren't wired up to the backend yet.

File: src/applications/disability-benefits/view-payments/actions/index.js (hunk 1 above)
Comment (cohnjesse) [reply src/applications/disability-benefits/view-payments/actions/index.js:26]: duh, updated this

File: src/applications/disability-benefits/view-payments/actions/index.js (hunk 2 above)
Comment (cohnjesse) [reply src/applications/disability-benefits/view-payments/actions/index.js:27]: on rated disabilities, we use the error code concatenated with `_internal_error` unless Im looking in the wrong place?

Comment (micahchiang) [reply src/applications/disability-benefits/view-payments/actions/index.js:27]: right, so here we'd probably want to do the same thing, using a word more pertinent to the error than `description`. For example `503_server_error` or `404_client_error`

Comment (cohnjesse) [reply src/applications/disability-benefits/view-payments/actions/index.js:27]: werd, just added this.

File: src/applications/disability-benefits/view-payments/actions/index.js [hunk 3]
```diff
@@ -16,11 +19,22 @@ function resolveAfter2Seconds() {
...
-    //   isServerError(errCode) ? recordEvent({}) : recordEvent({})
+    if (isServerError(response.errors[0].status)) {
+      recordEvent({
+        event: `disability-view-dependents-load-failed`,
+        'error-key': `${response.errors[0].status}_503_server_error`,
+      });
+    } else if (isClientError(response.errors)) {
+      recordEvent({
+        event: `disability-view-dependents-load-failed`,
+        'error-key': `${response.errors[0].status}_404_client_error`,
```
[the remark this answers is not in the document]
Comment (cohnjesse) [reply src/applications/disability-benefits/view-payments/actions/index.js:30]: crap, Atom reverted this. Should be updated now.

File: src/applications/disability-benefits/view-payments/actions/index.js [hunk 4]
```diff
@@ -16,11 +19,22 @@ function resolveAfter2Seconds() {
 export const getAllPayments = () => async dispatch => {
   const response = await resolveAfter2Seconds();
   if (response.errors) {
-    // TODO: fire off analytics event when endpoint is wired up.
-    //   const errCode = res.errors[0].code;
-    //   isServerError(errCode) ? recordEvent({}) : recordEvent({})
+    if (isServerError(response.errors[0].status)) {
+      recordEvent({
+        event: `disability-view-dependents-load-failed`,
+        'error-key': `${response.errors[0].status}_503_server_error`,
```
[the remark this answers is not in the document]
Comment (cohnjesse) [reply src/applications/disability-benefits/view-payments/actions/index.js:25]: Ah, gotcha. Updated

File: src/applications/disability-benefits/view-payments/actions/index.js [hunk 5]
```diff
@@ -16,11 +19,22 @@ function resolveAfter2Seconds() {
 export const getAllPayments = () => async dispatch => {
   const response = await resolveAfter2Seconds();
   if (response.errors) {
-    // TODO: fire off analytics event when endpoint is wired up.
-    //   const errCode = res.errors[0].code;
-    //   isServerError(errCode) ? recordEvent({}) : recordEvent({})
+    if (isServerError(response.errors[0].status)) {
+      recordEvent({
+        event: `disability-view-dependents-load-failed`,
```
[the remark this answers is not in the document]
Comment (cohnjesse) [reply src/applications/disability-benefits/view-payments/actions/index.js:24]: same as above

**Review by micahchiang — APPROVED** [review]
lgtm

**Pull request merged**